include/mbgl/storage/default_file_source.hpp

```diff
@@ -257,6 +257,7 @@ class DefaultFileSource : public FileSource {
 
     // For testing only.
     void setOnlineStatus(bool);
+    void reopenDatabaseReadOnlyForTesting();
```

Comment:
In [15768](https://github.com/mapbox/mapbox-gl-native/pull/15768/commits/8715a2e9a212e931ed132a49e37db52a97b3578a#diff-b92215f8737eee1d52ba8f24984bc27aR616) I tried to hide 'testing' APIs from public interface, would that work for `reopenDatabaseReadOnly`?

Reply:
I will hide testing API under generic setter once this PR is landed.